highway.tex

```diff
@@ -615,9 +630,201 @@ \subsection{Checkpoints}
+
+In a proof of work system, it is usually easy to control the amount of work that needs to go into processing and transmitting new messages. This is because a new block can be produced only at great computational cost, usually enough that it will take substantially fewer resources to process and propagate this block than were needed to produce it in the first place. However, in a proof-of-stake system, this no longer needs to be true. In fact, one could imagine an adversary trying to flood the system with new blocks (which are now computationally easy to manufacture), hoping that some or all of the other validators will not be able to cope with the stream. This is a possible form of denial of service attack against the system.
+
+Unfortunately, it is non-trivial to deal with such attacks. One cannot simply ignore votes from a validator who is sending too many of them. For one, the validator who is originating the votes may not be the one who you are receiving them from. If these votes are contained in the justification of a block sent by a confederate, it may not be clear whether the confederate is actively assisting the denial of service attack or if they simply had a greater download capacity than you do.
+
+This problem is made worse by noting that the honest validators must essentially agree on which votes they are going to process. If validator $A$ downloads one of these spammy votes and incorporates it into one of their own justifications, other validators will not be able to process $A$'s vote until they have also processed these other messages. This is perhaps even worse if an attacker sends different sets of spam votes to each of the honest validators. If each honest validator includes a reasonable subset of these messages in their next block, it may not be apparent that the system is being overtaxed until they have a chance to look at each others' blocks and realize that they will have a huge number of new votes that need to be downloaded and processed before they can continue.
+
+In this section, we first outline two of possible spam patterns that creates very high amounts of messages, in order to show that without taking any preventive measures, the spam attacks can cause a resource depletion in validator nodes and hence possibly crash the whole protocol. Then, we show how using relatively simple techniques, it is possible to bound the amount of possible spam to the easily managable size. 
+
+\subsection{Classification of Basic Spam Patterns}
+
+In this subsection, we lay out two basic categories of spam attacks:
+
+In a vertical spam, the attacker does not produce any equivocations and merely produces a single long string of sequential messages. This could be done by simply setting their round exponent to an unrealistically small number. This is substantially the easier type of spam to deal with, but the only type that can be produced without the attacker equivocating, and thus allowing them to be readily punished for it.
```

Comment:
`This is substantially the easier type ` -> `This is substantially the easiest type `

Reply:
@AdamGagol :point_up: